Lib/test/inspect_fodder.py

```diff
@@ -91,3 +91,31 @@ def as_method_of(self, obj):
 
 custom_method = Callable().as_method_of(42)
 del Callable
+
+
+def decorator(*args):
+    def inner(func):
+        @functools.wraps(func)
+        def wrapper():
+            pass
+        return wrapper
+    return inner
+
+
+@decorator(dict(), 24)
+@decorator(dict(), 1)
+@decorator(dict(), collections.defaultdict(lambda: 1))
+@decorator(
+    (()),
+    collections.defaultdict(lambda: 1),
+    [(i, j) for i, j in enumerate(range(5))],
+)
```

Comment:
Can you add a test where an argument to `@decorator` is a string containing a paren?

Reply:
Sure thing, although it should not cause any issue since the string will be a separate token than "()".